services/src/main/java/org/keycloak/credential/UserCredentialStoreManager.java

```diff
@@ -291,17 +292,20 @@ public boolean isConfiguredLocally(RealmModel realm, UserModel user, String type
     @Override
     public CredentialValidationOutput authenticate(KeycloakSession session, RealmModel realm, CredentialInput input) {
         List<CredentialAuthentication> list = UserStorageManager.getEnabledStorageProviders(session, realm, CredentialAuthentication.class);
+        
+        CredentialValidationOutput output = null;
         for (CredentialAuthentication auth : list) {
             if (auth.supportsCredentialAuthenticationFor(input.getType())) {
-                CredentialValidationOutput output = auth.authenticate(realm, input);
-                if (output != null) return output;
+                output = auth.authenticate(realm, input, output);
+                if (output != null && !LDAPStorageProvider.isKerberosUsernameExists(output)) return output;
```

Comment:
There shouldn't be any reference to LDAP or Kerberos at UserCredentialStoreManager level.